lib/tasks/comment_tasks.rake

```diff
@@ -0,0 +1,40 @@
+namespace :comment do
+  desc "Syncs the Comments.approved column to the new Comments.spam column"
+  task(:sync_approved_to_spam, [:limit] => :environment) do |_t, args|
+    # DB admin can pass in an optional integer for the limit
+    limit = args&.limit&.to_i
+
+    # success_count is used to provide status updates as the rake task is running.
+    success_count = 0
+
+    # Count is only used when a limit is provided. Otherwise, it is ignored.
+    count = 0 if limit.present?
+
+    # The approved and spam columns are inverse values of each other. If approved is true, then spam
+    # should be false, and vice versa. Thus, only records where approved and spam values are the same need to be synced.
+    Comment.where("(approved IS TRUE AND spam IS TRUE) OR (approved IS FALSE AND spam IS FALSE)").find_in_batches do |batch|
```

Comment:
This query will kill the database because there is no index on either of these columns, so it needs to retrieve all rows to check the where condition.

Instead, it would be better to simply get all comments and then just update all of them with the find_in_batches